### PR TITLE
Redirect to original workqueue when actioned directly from workqueue

### PR DESCRIPTION
**Pull request by jamil314**
## Description

Clearly describe what has been changed. Include relevant context or background.
Explain how the issue was fixed (if applicable) and the root cause.

Link this pull request to the GitHub issue (and optionally name the branch `ocrvs-<issue #>`)

## Checklist

- [ ] I have linked the correct Github issue under "Development"
- [ ] I have tested the changes locally, and written appropriate tests
- [ ] I have tested beyond the happy path (e.g. edge cases, failure paths)
- [ ] I have updated the changelog with this change (if applicable)
- [ ] I have updated the GitHub issue status accordingly

### CODE REVIEW

File: packages/client/src/v2-events/features/events/AdvancedSearch/SearchResult.tsx [hunk 1]
```diff
@@ -173,14 +181,59 @@ const ExtendedEventStatuses = {
...
+  let config: ActionConfig | undefined
+  if (actionType === 'DEFAULT') {
+    config = actionMenuItems.find(
+      ({ type }) =>
+        !(
+          [
+            ActionType.ASSIGN,
+            ActionType.UNASSIGN,
+            ActionType.READ
+          ] as readonly string[]
```
Comment (cibelius) [inline packages/client/src/v2-events/features/events/AdvancedSearch/SearchResult.tsx]: This same array appears in many place already I think, e.g. here: https://github.com/opencrvs/opencrvs-core/blob/develop/packages/commons/src/events/scopes.ts#L122-L127

Maybe its time to save it to constants somewhere? For example something like:

```
const META_ACTIONS: ActionType[] = [
  ActionType.ASSIGN,
  ActionType.UNASSIGN,
  ActionType.READ
]

function isMetaAction(actionType: ActionType) {
  return META_ACTIONS.includes(actionType)
}
```

With this we don't need to cast anymore:

```
  const config =
    actionType === 'DEFAULT'
      ? actionMenuItems.find(({ type }) => !isMetaAction(type))
      : configs[actionType]
```

File: packages/client/src/v2-events/features/events/AdvancedSearch/SearchResult.tsx [hunk 2]
```diff
@@ -227,6 +281,14 @@ export const SearchResultComponent = ({
   ) => {
     return eventData
       .map((event) => {
```
Comment (cibelius) [inline packages/client/src/v2-events/features/events/AdvancedSearch/SearchResult.tsx]: I think we can refactor away the double map, something like this:

```
  const mapEventsToWorkqueueRows = (
    eventData: (EventIndex & {
      title: string | null
      useFallbackTitle: boolean
    })[]
  ) => {
    return eventData.map((event) => {
      const actionConfigs = actions.map((actionType) => ({
        actionComponent: (
          <ActionComponent actionType={actionType} event={event} />
        )
      }))

      const eventConfig = eventConfigs.find(({ id }) => id === event.type)
      if (!eventConfig) {
        throw new Error('Event configuration not found for event:' + event.type)
      }

      const isInOutbox = outbox.some(
        (outboxEvent) => outboxEvent.id === event.id
      )
      const isInDrafts = drafts.some((draft) => draft.id === event.id)

      const getEventStatus = () => {
        if (isInOutbox) {
          return ExtendedEventStatuses.OUTBOX
        }
        if (isInDrafts) {
          return ExtendedEventStatuses.DRAFT
        }
        return event.status
      }

      const status = getEventStatus()
      return {
        actions: actionConfigs,
        label: eventConfig.label,
        type: intl.formatMessage(eventConfig.label),
        createdAt: formattedDuration(new Date(event.createdAt)),
        updatedAt: formattedDuration(new Date(event.updatedAt)),
        status: intl.formatMessage(messages.eventStatus, {
          status
        }),
        title: isInOutbox ? (
          <IconWithName name={event.title} status={status} />
        ) : (
          <TextButton
            color={event.useFallbackTitle ? 'red' : 'primary'}
            onClick={() => {
              return navigate(
                ROUTES.V2.EVENTS.OVERVIEW.buildPath({
                  eventId: event.id
                })
              )
            }}
          >
            <IconWithName name={event.title} status={status} />
          </TextButton>
        )
      }
    })
  }
```

I think it can be refactored further too. Please test that this works, I didnt test it thoroughly!

File: packages/client/src/v2-events/features/events/actions/declare/Pages.tsx [hunk 3]
```diff
@@ -93,7 +93,9 @@ export function Pages() {
       onSaveAndExit={async () =>
         handleSaveAndExit(() => {
           drafts.submitLocalDraft()
-          goToHome()
+          searchParams.workqueue
```
Comment (cibelius) [inline packages/client/src/v2-events/features/events/actions/declare/Pages.tsx]: I want to make sure that we want to hold the workqueue state in the search params, instead of just in react state for example?

File: packages/client/src/v2-events/features/events/actions/register/Review.tsx [hunk 4]
```diff
@@ -146,8 +153,7 @@ export function Review() {
         transactionId: uuid(),
         annotation
       })
-
-      goToHome()
+      slug ? goToWorkqueue(slug) : goToHome()
```
Comment (cibelius) [inline packages/client/src/v2-events/features/events/actions/register/Review.tsx]: We could make a small helper functions instead of doing this same logic everywhere, something like:

```
  function goToWorkqueueOrHome(slug?: string) {
    if (slug) {
      goToWorkqueue(slug)
    } else {
      goToHome()
    }
  }
```

File: packages/components/src/Workqueue/components/ListItemAction.tsx [hunk 5]
```diff
@@ -62,7 +62,7 @@ const ArrowExpansionSecion = styled(ArrowExpansionButton)<{
   ${({ isFullHeight }) => isFullHeight && ` height: 100%;`}
 `
 const ActionButtonContainer = styled.div`
-  width: 40px;
+  /* width: 40px; */
```
Comment (cibelius) [inline packages/components/src/Workqueue/components/ListItemAction.tsx]: oops

**Review by cibelius — APPROVED** [review]
Excellent work 💯
